From f81cde895534754f0bdbde147957883ce03291ec Mon Sep 17 00:00:00 2001
From: suchit-7x <gtsuchit7x@gmail.com>
Date: Mon, 24 Aug 2026 09:24:14 +0000
Subject: [PATCH] Skip openVTCamera call when VT camera id is -1

* mBackVtCameraId/mFrontVtCameraId are always "-1" on this device
* openCamera("-1", ...) throws IllegalArgumentException, uncaught
  by the existing CameraAccessException-only catch block, crashing
  the JpegEncoderThread and killing the whole camera process a few
  seconds after every capture
* skip the doomed call and release the wakelock instead, leaving
  checkConditionIsReady() and the rest of the reprocessing flow
  completely untouched
---
 .../camera/imagecodec/impl/VirtualCameraReprocessor.smali  | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/smali_classes5/com/xiaomi/camera/imagecodec/impl/VirtualCameraReprocessor.smali b/smali_classes5/com/xiaomi/camera/imagecodec/impl/VirtualCameraReprocessor.smali
index 74b552a..b067907 100644
--- a/smali_classes5/com/xiaomi/camera/imagecodec/impl/VirtualCameraReprocessor.smali
+++ b/smali_classes5/com/xiaomi/camera/imagecodec/impl/VirtualCameraReprocessor.smali
@@ -15,6 +15,13 @@
     invoke-virtual {v1}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
     move-result-object v1
     invoke-static {v0, v1}, Lcom/xiaomi/engine/Log;->i(Ljava/lang/String;Ljava/lang/String;)I
+    const-string v0, "-1"
+    invoke-virtual {v0, p1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+    move-result v0
+    if-eqz v0, :cond_vtcam_valid
+    invoke-direct {p0}, Lcom/xiaomi/camera/imagecodec/impl/VirtualCameraReprocessor;->releaseWakeLock()V
+    return-void
+    :cond_vtcam_valid
     :try_start_0
     iget-object v0, p0, Lcom/xiaomi/camera/imagecodec/impl/VirtualCameraReprocessor;->mCameraManager:Landroid/hardware/camera2/CameraManager;
     new-instance v1, Lcom/xiaomi/camera/imagecodec/impl/VirtualCameraReprocessor$3;
-- 
2.43.0

